Applications/Spire/Source/Login/LoginWindow.cpp

```diff
@@ -136,25 +142,19 @@ LoginWindow::LoginWindow(const std::string& version, QWidget* parent)
   auto password_layout = new QHBoxLayout();
   password_layout->setContentsMargins({});
   password_layout->setSpacing(0);
-  m_password_line_edit = new QLineEdit(this);
-  connect(m_password_line_edit, &QLineEdit::textEdited,
-    [=] {on_input_updated();});
-  connect(m_password_line_edit, &QLineEdit::textEdited,
-    [=] {on_password_updated();});
-  m_password_line_edit->setEchoMode(QLineEdit::Password);
-  m_password_line_edit->setPlaceholderText(tr("Password"));
-  m_password_line_edit->setSizePolicy(QSizePolicy::Expanding,
+  m_password_text_box = new TextBox(this);
+  set_style(*m_password_text_box, PASSWORD_INPUT_STYLE(get_style(
+    *m_password_text_box)));
+  m_password_text_box->get_model()->connect_current_signal(
+    [=] (const auto& current) {
+      on_input_updated();
+      on_password_updated();
```

Comment:
The current is already here, so I think the value can be passed into the on_password_updated function or the function can be replaced with m_chroma_hash_widget->set_text(current).

Reply:
I think we can take it a step further and remove both of the on_x_update() functions. The password has no effect on the condition in the on_input_updated() function, and there doesn't appear to be a case where editing the password will update the username. The statement of the on_password_updated() function can be put into the lambda, and the condition in the on_input_update() function can be simplified and put into the lambda slot for the username change signal.